function_computer_grid.py

```diff
@@ -0,0 +1,134 @@
+"""
+Created on Wed May 13 13:16:43 2020
+
+@author: carolinstreitberger
+"""
+import numpy as np
+import random
+
+def generate_random_position(size_ship):
+    """
+    This function will generate the position of a ship of a given size.
+
+    Parameters
+    ----------
+    size_ship : TYPE #?
+        Size of the ship whose position will be determined.
+
+    Returns
+    -------
+    orientation : TYPE
+        Vertical or horizontal.
+    y_start_position : TYPE
+        Starting point of the ship on the y-axis.
+    x_start_position : TYPE
+        Starting point of the ship on the x-axis.
+    """
+    orientation = random.randrange(0, 2, 1) # 0 is vertical and 1 is horizontal orientation
+    if (orientation == 0):
+        y_start_position = random.randrange(1, 11 - size_ship, 1)
+        # 11 - size_ship so that the ship is never bigger than 10
+        x_start_position = random.randrange(1, 11, 1)
+        # randrange is exclusive, ie 10 is never selected
+        # randrange is from 1 to 11 (so any number between 1 and 10) because 0 and 11 will be deleted later
+    elif (orientation == 1):
+        y_start_position = random.randrange(1, 11, 1)
+        x_start_position = random.randrange(1, 11 - size_ship, 1)
+    
+    return (orientation, y_start_position, x_start_position)
+
+def check_position(grid, size_ship, orientation, y_start_position, x_start_position):
+    """
+    This function will check if the position of the ship is valid.
+    So if the area including the ship and surrounding the ship on the grid is still free.
+
+    Parameters
+    ----------
+    grid : TYPE
+        The grid with the boats that are already positioned.
+    size_ship : TYPE
+        Size of the ship whose position will be checked.
+    orientation : TYPE
+        Vertical or horizontal.
+    y_start_position : TYPE
+        Starting point of the ship on the y-axis.
+    x_start_position : TYPE
+        Starting point of the ship on the x-axis.
+
+    Returns
+    -------
+    is_position_valid : TYPE
+        Variable to continue or stop the while loop (if True, the while loop is interrupted).
+    """
+    
+    if (orientation == 0):
+        y_to_check = slice((y_start_position - 1), (y_start_position + size_ship + 1), 1)
+        x_to_check = slice((x_start_position - 1), (x_start_position + 2), 1) # + 2 instead of + 1 because of python counting
+    elif (orientation == 1):
+        y_to_check = slice((y_start_position - 1), (y_start_position + 2), 1)
+        x_to_check = slice((x_start_position - 1), (x_start_position + size_ship + 1), 1)
+    
+    print("Part of the grid to be checked")
+    print(grid[y_to_check, x_to_check])
+    print("If these are all zeros then they will be added to the grid. Otherwise new positions will be generated")
+    
+    if np.any(grid[y_to_check, x_to_check] == 1):
+        is_position_valid = False
+    else:
+        is_position_valid = True
+        
+    return (is_position_valid)
```

Comment:
This if-else statement can be made a bit shorter. It is just a suggestion, and I am not sure whether it increases the readability. You could just state: return( not np.any(grid[y_to_check, x_to_check] == 1)). Because this itself already returns a True or False.

Reply:
If I implement this, then the function returns a True or False right? But I want the is_position_valid to be a True or False because it stops/ continues the while loop below (line 108: it either jumps back to 106 or 111). Would it do that too if I implement this suggestion?